test/lifecycle_test.dart

```diff
@@ -255,17 +264,22 @@ void main() {
 
       expect(component.lifecycleCalls, equals([
         matchCall('componentWillReceiveProps', args: [newPropsWithDefaults],           props: initialPropsWithDefaults, state: initialState),
-        matchCall('shouldComponentUpdate',     args: [newPropsWithDefaults, newState], props: initialPropsWithDefaults, state: initialState),
+        matchCall('componentWillReceivePropsWithContext', args: [newPropsWithDefaults, expectedContext],           props: initialPropsWithDefaults, state: initialState),
+        matchCall('shouldComponentUpdateWithContext',     args: [newPropsWithDefaults, newState, expectedContext], props: initialPropsWithDefaults, state: initialState),
         matchCall('componentWillUpdate',       args: [newPropsWithDefaults, newState], props: initialPropsWithDefaults, state: initialState),
+        matchCall('componentWillUpdateWithContext',       args: [newPropsWithDefaults, newState, expectedContext], props: initialPropsWithDefaults, state: initialState),
         matchCall('render',                                                                    props: newPropsWithDefaults, state: newState),
         matchCall('componentDidUpdate',        args: [initialPropsWithDefaults, initialState], props: newPropsWithDefaults, state: newState),
+        matchCall('componentDidUpdateWithContext',        args: [initialPropsWithDefaults, initialState, expectedContext], props: newPropsWithDefaults, state: newState),
       ]));
     });
 
-    group('when shouldComponentUpdate returns false:', () {
+    void testShouldUpdates({bool shouldComponentUpdateWithContext, bool shouldComponentUpdate}) {
       test('recieves updated props with correct lifecycle calls and does not rerender', () {
```

Comment:
There should be tests that context maps are updated as expected and with the correct values in the correct lifecycle methods.

An easy way to do that would be to parallel this props update test as well as the other "recieves updated props" test.